File: code/backend-php/src/Core/Domain/Auth/RefreshedSession.php
<?php

namespace Games\Paths\Core\Domain\Auth;

/**
 * Immutable domain model for the result of a token refresh operation.
 */
class RefreshedSession
{
    private string $userUuid;
    private string $username;
    private string $role;
    private string $accessToken;
    private string $refreshToken;
    private int $accessTokenExpiresAt;
    private int $refreshTokenExpiresAt;

    public function __construct(
        string $userUuid,
        string $username,
        string $role,
        string $accessToken,
        string $refreshToken,
        int $accessTokenExpiresAt,
        int $refreshTokenExpiresAt
    ) {
        $this->userUuid = $userUuid;
        $this->username = $username;
        $this->role = $role;
        $this->accessToken = $accessToken;
        $this->refreshToken = $refreshToken;
        $this->accessTokenExpiresAt = $accessTokenExpiresAt;
        $this->refreshTokenExpiresAt = $refreshTokenExpiresAt;
    }

    public function getUserUuid(): string
    {
        return $this->userUuid;
    }

    public function getUsername(): string
    {
        return $this->username;
    }

    public function getRole(): string
    {
        return $this->role;
    }

    public function getAccessToken(): string
    {
        return $this->accessToken;
    }

    public function getRefreshToken(): string
    {
        return $this->refreshToken;
    }

    public function getAccessTokenExpiresAt(): int
    {
        return $this->accessTokenExpiresAt;
    }

    public function getRefreshTokenExpiresAt(): int
    {
        return $this->refreshTokenExpiresAt;
    }
}

File: code/backend/java/core/src/main/java/games/paths/core/model/auth/RefreshedSession.java
package games.paths.core.model.auth;

/**
 * RefreshedSession - Immutable domain model representing the result of a token refresh.
 * Contains the new access and refresh tokens issued after a successful rotation.
 */
public class RefreshedSession extends BaseSession {

    private final String role;

    private RefreshedSession(Builder builder) {
        super(builder.userUuid, builder.username, builder.accessToken,
              builder.refreshToken, builder.accessTokenExpiresAt,
              builder.refreshTokenExpiresAt);
        this.role = builder.role;
    }

    public String getRole() {
        return role;
    }

    @Override
    public String toString() {
        return "RefreshedSession{userUuid='" + getUserUuid() + "', username='" + getUsername() + "'}";
    }

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder extends BaseSession.BaseBuilder<Builder> {
        private String role;

        public Builder role(String role) {
            this.role = role;
            return this;
        }

        public RefreshedSession build() {
            if (userUuid == null || userUuid.isBlank()) {
                throw new IllegalStateException("userUuid is required");
            }
            if (accessToken == null || accessToken.isBlank()) {
                throw new IllegalStateException("accessToken is required");
            }
            return new RefreshedSession(this);
        }
    }
}
